src/laboratory/shelfobject/serializers.py

```diff
@@ -20,3 +22,15 @@ class TransferOutShelfObjectSerializer(serializers.Serializer):
     amount_to_transfer = serializers.FloatField()
     mark_as_discard = serializers.BooleanField()
     laboratory = serializers.PrimaryKeyRelatedField(queryset=Laboratory.objects.all())
+
+
+class ShelfObjectDeleteSerializer(serializers.Serializer):
+    shelfobj = serializers.PrimaryKeyRelatedField(queryset=ShelfObject.objects.all())
+
+    def validate_shelfobj(self, value):
+        attr = super().validate(value)
+        if attr.in_where_laboratory_id != self.context.get('laboratory').pk:
+            logger.debug(f'attr.in_where_laboratory_id ({attr.in_where_laboratory_id}) '
+                         f'!= source_laboratory_id ({self.context.get("laboratory").pk})')
```

Comment:
like i said in the chat, in my case the variable is called source_laboratory_id, in your case it is not, please change it to actually have the right name according to what you use (laboratory).  Also, take a look at the latest version of the message we agreed on, it should include the name of the serializer.

Reply:
Here you can see what i did for transfer out https://github.com/Solvosoft/organilab/pull/230/files